atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -463,33 +464,34 @@ public Iterator<Map.Entry<Cell, byte[]>> getRowsColumnRange(
     }
 
     @Override
+    @SuppressWarnings("MustBeClosedChecker") // Sadly we can't close properly here without an ABI break :/
     public Map<byte[], Iterator<Map.Entry<Cell, byte[]>>> getRowsColumnRangeIterator(
             TableReference tableRef, Iterable<byte[]> rows, BatchColumnRangeSelection columnRangeSelection) {
         checkGetPreconditions(tableRef);
         if (Iterables.isEmpty(rows)) {
             return ImmutableMap.of();
         }
         hasReads = true;
-        Map<byte[], RowColumnRangeIterator> rawResults =
-                keyValueService.getRowsColumnRange(tableRef, rows, columnRangeSelection, getStartTimestamp());
-        ImmutableSortedMap.Builder<byte[], Iterator<Map.Entry<Cell, byte[]>>> postFilteredResultsBuilder =
-                ImmutableSortedMap.orderedBy(PtBytes.BYTES_COMPARATOR);
-        for (byte[] row : rows) {
-            Iterator<Entry<Cell, byte[]>> entryIterator = Optional.ofNullable(rawResults.get(row))
-                    .<Iterator<Entry<Cell, byte[]>>>map(rawIterator -> {
-                        // Sadly we can't close properly here without an ABI break :/
-                        @SuppressWarnings("MustBeClosedChecker")
-                        ClosableIterator<Entry<Cell, byte[]>> postFilteredIterator =
-                                getPostFilteredColumns(tableRef, columnRangeSelection, row, rawIterator);
-                        return postFilteredIterator;
-                    })
-                    .orElseGet(() -> getLocalWritesForColumnRange(tableRef, columnRangeSelection, row)
-                            .entrySet()
-                            .iterator());
-            postFilteredResultsBuilder.put(row, scopeToTransaction(entryIterator));
-        }
-        SortedMap<byte[], Iterator<Map.Entry<Cell, byte[]>>> postFilteredResults =
-                postFilteredResultsBuilder.buildOrThrow();
+        ImmutableSortedMap<byte[], RowColumnRangeIterator> rawResults = ImmutableSortedMap.copyOf(
+                keyValueService.getRowsColumnRange(tableRef, rows, columnRangeSelection, getStartTimestamp()),
+                PtBytes.BYTES_COMPARATOR);
```

Comment:
nice, thanks. Yeah in hindsight I forgot `Map<byte[], T>` for hash maps works in a pretty unpleasant way